src/Examine.Lucene/Search/LuceneSearchQuery.cs

```diff
@@ -97,20 +97,20 @@ public override IBooleanOperation RangeQuery<T>(string[] fields, T? min, T? max,
         protected override INestedBooleanOperation FieldNested<T>(string fieldName, T fieldValue)
             => RangeQueryInternal<T>(new[] { fieldName }, fieldValue, fieldValue, true, true, Occurrence);
 
-        protected override INestedBooleanOperation ManagedQueryNested(string query, string[] fields = null)
+        protected override INestedBooleanOperation ManagedQueryNested(string query, string[]? fields = null)
             => ManagedQueryInternal(query, fields, Occurrence);
 
         protected override INestedBooleanOperation RangeQueryNested<T>(string[] fields, T? min, T? max, bool minInclusive = true, bool maxInclusive = true)
             => RangeQueryInternal(fields, min, max, minInclusive, maxInclusive, Occurrence);
 
-        internal LuceneBooleanOperationBase ManagedQueryInternal(string query, string[] fields, Occur occurance)
+        internal LuceneBooleanOperationBase ManagedQueryInternal(string query, string[]? fields, Occur occurance)
         {
             Query.Add(new LateBoundQuery(() =>
             {
                 //if no fields are specified then use all fields
                 fields = fields ?? AllFields;
 
-                var types = fields.Select(f => _searchContext.GetFieldValueType(f)).Where(t => t != null);
+                var types = fields.Select(f => _searchContext.GetFieldValueType(f)).OfType<IIndexFieldValueType>();
```

Comment:
Is this changing logic?

Reply:
I don't think so no. `OfType` filters the items so that it's only items of the type that is in the list. So by using `IIndexFieldValueType` and not `IIndexFieldValueType?` we filter out every occurrence of `null`.

<img width="491" alt="image" src="https://github.com/Shazwazza/Examine/assets/24605285/4c52684f-22d0-4028-a302-f8a97cf5391a">

[Docs link](https://learn.microsoft.com/en-us/dotnet/api/system.linq.enumerable.oftype?view=net-7.0)

I've used this multiple times like this and never experienced a problem. So TLDR in this case it's just a way to do `Where(t => t != null)` but where the IDE picks up that all the null values have been removed.